fuse/opcode.go

```diff
@@ -177,6 +178,45 @@ func doWrite(server *Server, req *request) {
 	req.status = status
 }
 
+func doNotifyReply(server *Server, req *request) {
+	reply := (*NotifyRetrieveIn)(req.inData)
+	server.retrieveMu.Lock()
+	reading := server.retrieveTab[reply.Unique]
+	delete(server.retrieveTab, reply.Unique)
+	server.retrieveMu.Unlock()
+
+	badf := func(level byte, format string, argv ...interface{}) {
```

Comment:
Could you please clarify? The whole `doNotifyReply` is newly introduced in this patch. When we have a notify reply message from the kernel there are various cases that could be wrong, e.g. if it was some mismatch between what kernel sent and what we are expecting to receive. If that is the case we should somehow report the problem - log it. When logging from under notify reply handler there are several log messages possible, but they all have the same prefix. I just factored such logging with the same prefix into internal to doNotifyReply function. The level is also there because one badness condition is not too bad - e.g. if the kernel sent us more data than we requested, we can ignore the tail but still use the requested part. Still it was some error logic, probably on the kernel side, to send more data, and it is better for it to not to go unnoticed. That's why badf also accepts the prefix.

In other words here it is mostly syntatic sugar not to duplicate the error prefix.

Do you want me to remove badf from this patch and just use log.Printf directly?

Reply:
I'm talking about just the logging. I introduces log levels (errors, warning) with abbreviations (E, W), which aren't used in other places. I think it's good to introduce some more structure into the log messages, but would rather see a change that addresses all log statements throughout the code.

Reply:
Ok, I'm dropping log levels here.